src/libcommonserver/io/iohelper.h

```diff
@@ -25,7 +25,34 @@ namespace KDC {
 
 struct FileStat;
 
+class DirectoryIterator {
+    public:
+        DirectoryIterator(const SyncPath &directoryPath, bool recursive, IoError &ioError,
+                          DirectoryOptions option = DirectoryOptions::none);
+
+        DirectoryIterator() = default;
+        //! Get the next directory entry.
+        /*!
+          \param nextEntry is set with the next directory entry.
+          \param ioError holds the error returned when an underlying OS API call fails (e.g., IoErrorEndOfDirectory,
+          IoErrorInvalidDirectoryiterrator). \return true if no unexpected error occurred, false otherwise. See _isExpectedError
```

Comment:
```suggestion
          IoErrorInvalidDirectoryiterrator). \return true if no unexpected error occurred, false otherwise. See IoError::_isExpectedError
```
Should `IoError::_isExpectedError` be made public?

Reply:
In the latest commit, IoHelper::next does not rely on this function anymore.